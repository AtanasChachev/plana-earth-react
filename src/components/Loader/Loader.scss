.loader {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: var(--loader-background);
  border-radius: 4px;
  visibility: hidden;
  opacity: 0;
  transform: scale(0);
  transition: transform .4s, opacity .4s, visibility .4s;

  &--visible {
    visibility: visible;
    opacity: 1;
    transform: scale(1);
  }
}