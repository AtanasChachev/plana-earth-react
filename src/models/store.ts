import { ChartState } from './chart';
import { ProductState } from './products';

export type State = {
  productsState: ProductState;
  chartState: ChartState;
};

export type StateAction = {
  type: symbol;
  payload: never;
};