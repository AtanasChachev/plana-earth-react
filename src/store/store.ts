import { createStore, combineReducers } from 'redux';
import ProductsReducer from './reducers/products';

const reducers = combineReducers({
  products: ProductsReducer,
});

export const store = createStore(reducers);
