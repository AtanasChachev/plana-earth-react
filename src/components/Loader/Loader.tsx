import CircularProgress from '@material-ui/core/CircularProgress';
import './Loader.scss';

interface LoaderProps {
  showLoader?: boolean;
}

const Loader = ({ showLoader }: LoaderProps): JSX.Element => 
  <div className={`loader ${showLoader ? 'loader--visible' : ''}`}>
      <CircularProgress size={100} className="loader__spinner" color='inherit' />
  </div>;

export { Loader };