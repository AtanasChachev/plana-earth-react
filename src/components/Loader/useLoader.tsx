import { State } from 'models/store';
import { UserInterfaceState } from 'models/ui';
import { useSelector } from 'react-redux';

type Return = {
  shShowLoader: boolean;
};

export const useLoader = (): Return => {
  const { shShowLoader }: UserInterfaceState = useSelector((state: State) => state.uiState);

  return {
    shShowLoader,
  };
};