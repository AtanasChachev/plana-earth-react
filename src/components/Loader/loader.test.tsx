import { render } from '@testing-library/react';
import { Loader } from './Loader';
import { Provider } from 'react-redux';
import { store } from 'store/store';

const setup = () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Loader />
    </Provider>,
  );

  return {
    asFragment,
  };
};

describe('<Loader />', () => {
  test('<Loader /> renders correctly', () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });
});