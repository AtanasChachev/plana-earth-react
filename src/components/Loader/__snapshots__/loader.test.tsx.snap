// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> <Loader /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="loader "
  >
    <div
      class="MuiCircularProgress-root loader__spinner MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 60px; height: 60px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
